server.js

```diff
@@ -9,7 +9,7 @@ const { connectToDB } = require('./helpers/db');
 
 const app = express();
 const server = http.createServer(app);
-const port = process.env.PORT || 4000;
+const { PORT = 4000 } = process.env;
```

Comment:
Can anyone tell me what is happening on this line?

Reply:
I think you used the ES6 shorter, cleaner and more understandable version of Node.js

Reply:
Normally i think this is how the process object looks like.
process = {
      env: { 
          PORT: default_value_if_there_is
      }
   }
So, const { PORT = 4000 } = process.env;
will set PORT to 4000 even if there is something in the environment variable PORT.
In general i think the principle applied here is destructuring assignment syntax

Reply:
Wow, that's superb.
so it does exactly same as this `const port = process.env.PORT || 4000;`

Reply:
It does almost the same thing. `const { PORT = 4000 } = process.env;` sets a default value of 4000 on `PORT` only if `PORT` is `undefined`. Whereas `const port = process.env.PORT || 4000;` sets port to 4000 if `PORT` is null, false, 0, ""/'', or undefined.